### ISSUE TITLE
Compiling error with AP_ADSB snprintf

### ISSUE THREAD

**Issue by hoangthien94** [ArduPilot/ardupilot#8947]
### Bug report

**Issue details**

When I tried to compile the code with `waf`, I got the error `error: 'snprintf' output may be truncated before the last format character [-Werror=format-truncation=]`. 

Maybe the issue is related to this PR https://github.com/ArduPilot/ardupilot/pull/8497?

**Version**

ArduCopter on master, commit 9cba2c9bd3b98895ffad546f047db270923e6092

**Platform**
[  ] All
[  ] AntennaTracker
[x] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Logs**
```
$ ./waf configure --board px4-v3
$ ./waf copter
...
[439/521] Compiling libraries/AC_AttitudeControl/AC_PosControl.cpp
../../libraries/AP_ADSB/AP_ADSB.cpp: In member function 'uint8_t AP_ADSB::get_encoded_callsign_null_char()':
../../libraries/AP_ADSB/AP_ADSB.cpp:594:9: error: 'snprintf' output may be truncated before the last format character [-Werror=format-truncation=]
 uint8_t AP_ADSB::get_encoded_callsign_null_char()
         ^~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
```

**Comment by hoangthien94** [comment]
Change [this line](https://github.com/AzurDrones/ardupilot/blob/eac2bd88be1596825c739fccb00b4cb9bf29f339/libraries/AP_ADSB/AP_ADSB.cpp#L642 ) to this fixed the error:

`
snprintf(out_state.cfg.callsign, 5, "%04d", unsigned(out_state.cfg.squawk_octal) & 0x001F);
`

**Comment by rmackay9** [comment]
@hoangthien94, I tried compiling using waf on WSL on windows and hit a different error :-).  I'd say PR that change and hopefully we can pull it in.

**Comment by magicrub** [comment]
I actually fixed that for one compiler and it then broke another compiler. Let's make sure whatever fix we do actually fixes it :)

**Comment by hoangthien94** [comment]
Implemented in https://github.com/ArduPilot/ardupilot/pull/8973. @rmackay9 please try to see if it can be pulled in :)

**Comment by Raviu8** [comment]
Same issue here; It's failed on all px4-v? boards.
```
../../libraries/AP_ADSB/AP_ADSB.cpp: In member function 'uint8_t AP_ADSB::get_encoded_callsign_null_char()':
../../libraries/AP_ADSB/AP_ADSB.cpp:594:9: error: 'snprintf' output may be truncated before the last format character [-Werror=format-truncation=]
 uint8_t AP_ADSB::get_encoded_callsign_null_char()
         ^~~~~~~
compilation terminated due to -Wfatal-errors.
cc1plus: all warnings being treated as errors
```

Compiler is: gcc-arm-none-eabi-7-2017-q4-major

But on Pixhawk1 board it's treated as warning and doesn't fail

```
../../libraries/AP_ADSB/AP_ADSB.cpp: In member function 'uint8_t AP_ADSB::get_encoded_callsign_null_char()':
../../libraries/AP_ADSB/AP_ADSB.cpp:594:9: warning: 'snprintf' output may be truncated before the last format character [-Wformat-truncation=]
 uint8_t AP_ADSB::get_encoded_callsign_null_char()
         ^~~~~~~
../../libraries/AP_ADSB/AP_ADSB.cpp:642:13: note: 'snprintf' output between 5 and 6 bytes into a destination of size 5
     snprintf(out_state.cfg.callsign, 5, "%04d", unsigned(out_state.cfg.squawk_octal));
     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Changing the size in mentioned line to 6 allows compilation on px4 boards, but I have no idea if the change is correct.

```
snprintf(out_state.cfg.callsign, 6, "%04d", unsigned(out_state.cfg.squawk_octal));
``` 

By the way, Shouldn't warnings on Pixhawk* boards be treated as errors too?

Side note: Almost nothing could be compiled on the newest compiler/toolchain from arm gcc-arm-none-eabi-7-2018-q2-update-linux ; On px4-v? boards Nuttx failed to compile, on Pixhawk? chibios compiled but ardupilot fails with many errors. Should I open a separate bug?

**Comment by OXINARF** [comment]
Fixed with #8973.
